### PR TITLE
Use Context.context in command execution for funcbench

### PR DESCRIPTION
**Pull request by nevill**
1. Use context to kill command, it finishes @bwplotka 's TODO
~2. Format the benchmark result, add PR number in the column header, see @geekodour 's comments in #350.~

### CODE REVIEW

**Review by geekodour — COMMENTED** [review]
comment about tests. other changes LGTM.

**Comment by geekodour** [comment]
I've another format in mind:

Instead of :
| Benchmark | Old ns/op | PR-54 ns/op | Delta |
|-|-|-|-|
BenchmarkBufferedSeriesIterator-8|15.7|15.7|+0.00%

We could also do:
**Old:** `master`/`9a21fdcd1b787b618f1f946e3434391bc5e6271b`
**New:** `PR-3455`
| Benchmark | Old ns/op | New ns/op | Delta |
|-|-|-|-|
BenchmarkBufferedSeriesIterator-8|15.7|15.7|+0.00%

This way we keep the markdown formatter independent of being run on a PR, we can write a wrapper function that adds the **Old** and **New** legends to it.

Would like some feedback on this one.

File: funcbench/env.go [hunk 1]
```diff
@@ -128,21 +129,55 @@ func newGitHubEnv(ctx context.Context, e environment, gc *gitHubClient, workspac
...
+func (g *GitHub) PostResults(ctx context.Context, cmps []BenchCmp) error {
 	b := bytes.Buffer{}
 	Render(&b, cmps, false, false, g.compareTarget)
-	return g.client.postComment(formatCommentToMD(b.String()))
+	return g.client.postComment(ctx, g.formatCommentToMD(b.String()))
 }
 
 func (g *GitHub) Repo() *git.Repository { return g.repo }
 
+func (g *GitHub) formatCommentToMD(rawTable string) string {
```
[the remark this answers is not in the document]
Comment (nevill) [reply funcbench/env.go]: Is there any test of `funcbench` running by CircleCI? Any links?

[the remark this answers is not in the document]
Comment (nevill) [reply funcbench/env.go]: Oh, I guess you mean `main_test.go`.

**Comment by nevill** [comment]
I like the idea to have a legend.
We can keep current implementation `formatCommentToMD` in `bench.go` for now, though the test being in `main_test.go` is a bit strange.

A better idea is to have a `RenderToMarkDown` in `cmp.go`, like `Render` there. Even implement a `FormatMarkDown`, if we switch to `benchstat`.

**Comment by nevill** [comment]
Somewhere we mentioned that we need a kind of summary of the current funcbench result, otherwise it's too long to check.
A sample is [here](https://github.com/nevill/prometheus/pull/3#issuecomment-615084625).

File: funcbench/bench.go [hunk 2]
```diff
@@ -67,7 +68,7 @@ func (b *Benchmarker) benchOutFileName(commit plumbing.Hash) (string, error) {
 	return fmt.Sprintf("%s-%s.out", bb.String(), commit.String()), nil
 }
 
-func (b *Benchmarker) execBenchmark(pkgRoot string, commit plumbing.Hash) (out string, err error) {
+func (b *Benchmarker) execBenchmark(ctx context.Context, pkgRoot string, commit plumbing.Hash) (out string, err error) {
```
Comment (bwplotka) [inline funcbench/bench.go]:
```suggestion
func (b *Benchmarker) exec(ctx context.Context, pkgRoot string, commit plumbing.Hash) (out string, err error) {
```
We are in benchmark so it's kind of clear. Yes. I know I did in the first place (: But good ideas come with time.

**Review by bwplotka — APPROVED** [review]
Cool thanks, LGTM just please **Always have meaningful commit name and PR Title** 

This has to be explanatory (:

**Comment by geekodour** [comment]
> Somewhere we mentioned that we need a kind of summary of the current funcbench result, otherwise it's too long to check.

@nevill let's keep the changes to the comment to another PR as it seems some more thought will need to be given to the formatting. We could possibly use collapsible sections. (https://github.com/dear-github/dear-github/issues/166#issuecomment-236342209)

> We can keep current implementation `formatCommentToMD` in bench.go for now, though the test being in main_test.go is a bit strange. A better idea is to have a RenderToMarkDown in cmp.go, like Render there.

Agreed with the suggestion for the file change for the test and function `formatCommentToMD` and yes, `RenderToMarkDown` does seem like a better name.

**Comment by nevill** [comment]
@geekodour I removed the changes to `formatCommentToMD`, it's ready to merge.

**Comment by geekodour** [comment]
Thanks!

**Pull request merged**